lib/catalog.ex

```diff
@@ -127,6 +118,18 @@ defmodule Catalog do
     |> Enum.to_list()
   end
 
+  @spec populate_ets_table(:ets.tid(), [t()]) :: [t()]
+  defp populate_ets_table(ets, documents) do
+    for document <- documents do
+      _ =
+        :ets.insert(
+          ets,
+          {Document.key(document), document.last_name, document.first_name,
+           document.date_of_birth, document}
+        )
+    end
```

Comment:
`:ets.insert` can also take a list, so you could do a map and then insert the documents all at once:
```elixir
records = for document <- documents do
 {Document.key(document), document.last_name, document.first_name, document.date_of_birth, document}
end
:ets.insert(ets, records)
```